### DISCUSSION TITLE
DBT json log file

### DISCUSSION THREAD

**Discussion by shanebusenitz** [dagster-io/dagster#24004] [Q&A]
I'm running several Assets with:
```
DbtCliResource(
            project_dir= DBT_PROJECT_PATH,
            profiles_dir= DBT_PROFILES,
            global_config_flags=["--log-format-file", "json"]
    )
```
and then:
```
@dbt_assets(
    manifest=MANIFEST,
    dagster_dbt_translator=CustomDagsterDbtTranslator()
)
def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
    dbt_build_invocation = dbt.cli(["build"], context=context)

    yield from dbt_build_invocation.stream().fetch_row_counts().fetch_column_metadata()
```

I'm using the `S3ComputeLogManager` to log to S3 however that appears to just log the `stderr` and `stdout`. How would I get the JSON `logs/dbt.log` file to S3 as well?

**Comment by garethbrickman** [comment]
I think you may be able to use [DbtCliInvocation. get_artifact ](https://docs.dagster.io/_apidocs/libraries/dagster-dbt#dagster_dbt.DbtCliInvocation.get_artifact) to fetch the `dbt.log`. You could then either log its contents into the run logs ([like in this example](https://github.com/dagster-io/hooli-data-eng-pipelines/blob/8c5909e17e89af22f51bf6d5bfccd8e1c4eb310a/hooli_data_eng/assets/dbt_assets.py#L210C51-L210C67)), or use a separate process to upload it entirely to S3.
